doc/user/path.rst

```diff
@@ -0,0 +1,415 @@
+.. _path:
+
+****
+PATH
+****
+
+:abbr:`PATH` is a daemon that handles the installation and deletion
+of Segment Routing (SR) Policies.
+
+
+.. _starting-path:
+
+Starting PATH
+=============
+
+Default configuration file for *pathd* is :file:`pathd.conf`.  The typical
+location of :file:`pathd.conf` is |INSTALL_PREFIX_ETC|/pathd.conf.
+
+If the user is using integrated config, then :file:`pathd.conf` need not be
+present and the :file:`frr.conf` is read instead.
+
+.. program:: pathd
+
+:abbr:`PATH` supports all the common FRR daemon start options which are
+documented elsewhere.
+
+
+PCEP Support
+============
+
+To build the PCC for pathd, the `pceplib external library <https://github.com/volta-networks/pceplib>`_.
+The last commit of pceplib the module has been tested with is stored in
+the file `pathd/pceplib.lock`.
+The following extra parameter should be passed to `configure`:
+
+```
+--enable-pcep LDFLAGS="-L${PCEPLIB_ROOT}/install/lib" CPPFLAGS="-I${PCEPLIB_ROOT}/install/include"
+```
+
+To start pathd with pcep support the extra parameter `-M pathd_pcep` should be
+passed to the pathd daemon.
+
+
+.. _path-commands:
+
+
+Pathd Commands
+==============
+
+Configuration example:
+
+.. code-block:: frr
+
+  debug pathd pcep basic
+  pcep-config-group GROUP1
+   source-address 1.1.1.1
+   tcp-md5-auth secret
+   timer keep-alive 30
+  !
+  pcc-peer PCE1
+   config-group GROUP1
+   address ip 10.10.10.10
+  !
+  pcc-peer PCE2
+   config-group GROUP1
+   address ip 9.9.9.9
+  !
+  pcc
+   peer PCE1 precedence 10
+   peer PCE2 precedence 20
+  !
```

Comment:
I am confused because it looks to me that the `pce` command on cisco is to configure a PCE and it uses `mpls traffic-eng pcc ...` to configure a PCC. In our case we are configuring a PCC that connects to multiple PCE.

Reply:
In addition, we choosed `pcc-peer` instead of `pce` because we may have an actual PCE later on that would use the `pce` command.

Reply:
Cisco use `mpls traffic-eng pce peer ...` to configure the pcep session with the PCE. And Juniper used 'protocol pcep pce XXX` for the same purpose. Here, you are configured the PCEP session parameter to connect to the PCE i.e. the IP address and port of the PCE not of the PCC. So, `pcc-peer` must be change to `pce-peer`

Regarding PCE feature, you don't need to configure a peer. A PCE is a server waiting for incoming PCEP session coming from PCC.

Reply:
Hi all, Javier from Volta here.

Regarding this and what cisco (xr9k 6.5.1) uses , even also the `mpls traffic-eng pce peer ` exists ("[Configures a static PCE peer address](https://www.cisco.com/c/en/us/td/docs/routers/asr9000/software/asr9k-r6-2/mpls/configuration/guide/b-mpls-cg-asr9000-62x/b-mpls-cg60x-asr9k_chapter_0110.html)" ), the bellow sintax also exists.

- Config a cisco node also as pcc

`
segment-routing
 traffic-eng
  !
  pcc
   source-address ipv4 10.10.10.1
   pce address ipv4 10.10.10.1
   !
   report-all
  !

`

- So maybe we can change the pcc-peer section to show (the source already exists)  remote as cisco does adding `pce` to remote pce  address

`
 pcc-peer PCE2
   source-address ip 1.1.1.1
   pce address ip 9.9.9.9
   config-group GROUP1

`



We in volta use 
`RP/0/RP0/CPU0:R1#show version
Wed Nov 18 07:35:23.900 UTC
Cisco IOS XR Software, Version 6.5.1
Copyright (c) 2013-2018 by Cisco Systems, Inc.
`